fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalValuesOperator.java

```diff
@@ -70,12 +80,22 @@ public <R, C> R accept(OptExpressionVisitor<R, C> visitor, OptExpression optExpr
 
     @Override
     public boolean equals(Object o) {
-        return this == o;
```

Comment:
I think the `LogicalValuesOperator` has nothing differnent from other operators. In the tests, I found some cases are  infinite optimization because of the same `LogicalValuesOperaor` always being inserted into the memo. So I change its `equal() ` method.